cloud/services/domains.go

```diff
@@ -62,6 +74,74 @@ func EnsureDNSEntries(ctx context.Context, mscope *scope.MachineScope, operation
 	return nil
 }
 
+// EnsureAkamaiDNSEntries ensures the domainrecord on Akamai EDGE DNS is created, updated, or deleted based on operation passed
+func EnsureAkamaiDNSEntries(ctx context.Context, mscope *scope.MachineScope, operation string, dnsEntries []DNSOptions) error {
+	// SetUp Client for Akamai EDGE DNS
+
+	fqdn := mscope.LinodeCluster.ObjectMeta.Name + "-" + mscope.LinodeCluster.Spec.Network.DNSUniqueIdentifier + "." + mscope.LinodeCluster.Spec.Network.DNSRootDomain
+
+	for _, dnsEntry := range dnsEntries {
+		recordBody, err := mscope.AkamaiDomainsClient.GetRecord(ctx, mscope.LinodeCluster.Spec.Network.DNSRootDomain, fqdn, string(dnsEntry.DNSRecordType))
+		if err != nil {
+			if operation == "create" {
+				if err := createAkamaiDNSEntry(ctx, mscope, fqdn, dnsEntry); err != nil {
+					return err
+				}
+			}
+			continue
+		}
+		if operation == "delete" {
+			switch {
+			case len(recordBody.Target) > 1:
```

Comment:
is this switch to avoid the nestif on the linter?

Reply:
unfortunately, yes